cinema/serializers.py

```diff
@@ -68,13 +71,52 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available",
         )
 
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        total_capacity = instance.cinema_hall.capacity
+        booked_tickets = instance.tickets.count()
+        representation["tickets_available"] = total_capacity - booked_tickets
+        return representation
+
 
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = serializers.SerializerMethodField()
+
+    def get_taken_places(self, obj):
+        return [
+            {"row": ticket.row, "seat": ticket.seat}
+            for ticket in obj.tickets.all()
+        ]
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    movie_session = MovieSessionListSerializer()
+
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(many=True, read_only=True)
```

Comment:
The `tickets` field in `OrderSerializer` should be marked as read-only to prevent unintended modifications during order creation or updates. This was highlighted in the previous review but hasn't been addressed.